packages/api/src/util/getProvider.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {assert} from '@cennznet/util';
+import {HttpProvider, WsProvider} from '@plugnet/rpc-provider';
+import {ProviderInterface} from '@plugnet/rpc-provider/types';
+
+const REGEX = /^(wss?|https?|WSS?|HTTPS?):\/\//;
```

Comment:
since you already convert the url string to lowercase, you can remove WSS?|HTTPS from regex